### ISSUE TITLE
Reverse lookup in Belgium

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#3430]
**[Submitted to the original trac issue database at 11.00pm, Monday, 3rd January 2011]**

Currently the url http://nominatim.openstreetmap.org/reverse?format=xml&lat=50.942496&lon=4.645713&zoom=18&addressdetails=1 gives as result:

<result place_id="125122431" osm_type="way" osm_id="87751721">
Peter De Brabandere, 1, Lipseweg, Tildonk, Haacht, Flemish Brabant, Flemish Community, Flanders, Belgium
</result>
<addressparts>
<house_number>1</house_number>
<pharmacy>Peter De Brabandere</pharmacy>
<road>Lipseweg</road>
<village>Tildonk</village>
<city>Haacht</city>
<county>Flemish Brabant</county>
<state_district>Flemish Community</state_district>
<state>Flanders</state>
<country>Belgium</country>
<country_code>be</country_code>
</addressparts>

I have various issues with that, but I currently don't have good solutions for all of it.
- "Haacht" is not an official city.  The node is mapped as a town, the relation has admin level 8.  An admin level of 8 for Belgium should be called "municipality" in English.   A municipality might contain a city in admin level 9, or it might not.
- Admin level 9 which isn't mapped would be a sub-municipality.
- "Flemish Brabant" is a province, not a county.  (admin level 6.)
- "Flemish Community" is a community, not a state district. (level 5)
- "Flanders" is a region, not a state. (level 4)

An official address in Belgium contains:
- Street and house number
- municipality
- zip code (which doesn't seem to be reported here, but does in other places, need to look at it)
- Country

People might also write the sub-municipality on an address, but it's not the official one.  We currently don't (always) map admin level 9 because it's not always clear where the borders are, because officially there are no borders, but most of should have a node indicating the centre of it.  If there is an admin level of 9 you could use it, but you can also stop at level 8.
When looking up names of places it is useful to look at the nodes within the admin level 8, but not for a reverse lookup.  For the reverse lookup I would stop using the nodes.

Now it's reporting all kinds of the middle admin levels that most people won't care about (4-6).  Some people might want to see what province this is in, but then things gets interesting for the Brussel Capital Region where the higher government for the municipality is not a province, since it doesn't have any, but the region.

Looking at Brussel centre I now get:
<result place_id="13274409" osm_type="way" osm_id="3517898">
Grand Place, Saint-Josse-ten-Noode - Sint-Joost-ten-Node, Ville de Bruxelles - Stad Brussel, 1000, Flemish Community, Brussels-Capital Region, Belgium
</result>
<addressparts>
<pedestrian>Grand Place</pedestrian>
<town>Saint-Josse-ten-Noode - Sint-Joost-ten-Node</town>
<city>Ville de Bruxelles - Stad Brussel</city>
<postcode>1000</postcode>
<state_district>Flemish Community</state_district>
<state>Brussels-Capital Region</state>
<country>Belgium</country>
<country_code>be</country_code>
</addressparts>

In this case the town is completely wrong, I'd guess because there is no node for Brussel itself.

But as you see there is no province (county).

The "Flemish Community" part is also wrong since it's part of both the Flemish and French community.

In short, I would only use admin level 8.

When writing an address we also use "street number", not "number, street".

**Comment by tomhughes** [comment]
**Author: kurt@roeckx.be**
**[Added to the original trac issue at 5.44pm, Tuesday, 4th January 2011]**

Thinking about this some more, maybe we should just skip admin level 5 (community, state_district).  It's a real administrative level, but it's at the same "level" as level 4.  Where level 4 is more about "land", level 5 is more about "people".

**Comment by tomhughes** [comment]
**Author: gerdami**
**[Added to the original trac issue at 6.21pm, Wednesday, 28th March 2012]**

A search for Bruxelles gives
Brussels, Flemish Community, Brussels-Capital Region, Belgium, Europe (City) (details)

While this is not wrong, this not totally true since Brussels belongs also to the French Community of Belgium.

Indeed there is a legal overlap of both communities over the Brussels-Capital Region, with NO hierarchy in any direction between communities and regions.

I you examine http://nominatim.openstreetmap.org/details.php?place_id=157050627
you will see that French Community is dimmed in paragraph 


== Address ==

Brussels (Type: place:city, node 1635651356, 15, , Polygon, 0 GOTO)

Flemish Community (Type: boundary:administrative, relation 53136, 5, , Polygon, 0.219284367884933 GOTO)

*French Community (Type: boundary:administrative, relation 78967, 5, , Polygon, 0.836828863064937 GOTO)*

Brussels-Capital Region (Type: boundary:administrative, relation 54094, 4, , Polygon, 0.0216817423783759 GOTO)

Belgium (Type: boundary:administrative, relation 52411, 2, , Polygon, 0.286100598523024 GOTO)


I agree with the post above, communities should NOT be displayed as part of the address.


However, one could consider the linguistic regions which do NOT overlap, based on Article 4 of the Belgian Constitution:

```
Article 4

     Belgium comprises four linguistic regions : the Dutch-speaking region, the French-speaking region, the bilingual region of Brussels-Capital and the German-speaking region.
     Each municipality of the Kingdom forms a part of one of these linguistic regions.
```
Source: http://www.const-court.be/en/basic_text/basic_text_constitution.html

**Comment by tomhughes** [comment]
**Author: lonvia**
**[Added to the original trac issue at 8.34pm, Wednesday, 4th February 2015]**

The issue has been solved in the meantime by retagging the language boundaries as boundary=political.
